README.md

```diff
@@ -669,6 +669,105 @@ you've fixed the error, it's usually safe to re-run the `spin-up` or `tear-down`
 for the modules that already deployed successfully, and should only affect the ones that had an error the last time
 around.
 
+## Remote Terraform configurations
+
+Terragrunt has the ability to work with Terraform configurations that are not in your working directory. This is 
+especially useful if you have versioned all of your Terraform code and want to `apply` a specific version of it
+(see [Using Pipelines to Manage Environments with Infrastructure as 
+Code](https://medium.com/@kief/https-medium-com-kief-using-pipelines-to-manage-environments-with-infrastructure-as-code-b37285a1cbf5)).
+
+In your `.terragrunt` file, you can specify a `terraform` block with a `source` parameter that tells Terragrunt where 
+to find the Terraform configuration files:
+
+```hcl
+terraform {
+  source = "git::git@github.com:foo/bar.git//my-terraform-configurations?ref=v0.0.1"
+}
+```
+
+When Terragrunt finds a `terraform` block, it will:
+ 
+1. Download the configurations specified via the `source` parameter into a temporary folder. This is done using the
+   [terraform init command](https://www.terraform.io/docs/commands/init.html), so the `source` parameter supports the
+   exact same syntax as the [module source](https://www.terraform.io/docs/modules/sources.html) parameter, including
+   local file paths, Git URLs, and Git URLs with `ref` parameters (useful for checking out a specific tag, commit, or
+   branch of Git repo).
+1. Copy all files from the current working directory into the temporary folder. This is useful if you have `.tfvars`
+   files in your local path that you are using to set variables within your templates.
+1. Execute whatever Terraform command you specified in that temporary folder. **Note**: if you are passing any file
+   paths (other than paths to files in the current working directory) to Terraform via command-line options, those 
+   paths must be absolute paths since we will be running Terraform from the temporary folder!
+
+With this approach, you could define all of your Terraform code in one repo called, for example, 
+infrastructure-modules:
+
+```
+infrastructure-modules
+  └ frontend-app
+      └ main.tf
+      └ vars.tf
+      └ outputs.tf
+  └ backend-app
+  └ search-app
+  └ mysql
+  └ redis
+  └ vpc
+```
```

Comment:
This isn't the ideal place for this comment, but it occurs to me that this infra-modules folder structure should use folder groupings like:

```
- services
  - frontend-app
    - main.tf
  - backend-app
- data-stores
  - mysql
  - redis
```